### PR TITLE
DSD-1589: Adds placeholder props and allows empty state

### PR DESCRIPTION
**Pull request by oliviawongnyc**
Fixes JIRA ticket [DSD-1589](https://jira.nypl.org/browse/DSD-1589). 

NOTE: this is a draft PR because Edwin and I made some assumptions about how this should work and want to get sign off from others on the approach before finishing the PR (adding to the Changelogs, updating Storybook, etc).

## This PR does the following:

- Allows users to pass an empty string for initialDate and initialDateTo, which results in empty inputs.
- Allows users to optionally pass a string placeholder and placeholderTo.

## How has this been tested?

- Locally, on Storybook.

## Accessibility concerns or updates

- None.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -275,8 +284,16 @@ export const DatePicker = chakra(
     } = props;
     const styles = useMultiStyleConfig("DatePicker", {});
     const finalStyles = isDateRange ? styles : {};
-    const initStartDate = initialDate ? new Date(initialDate) : new Date();
-    const initEndDate = initialDateTo ? new Date(initialDateTo) : new Date();
+    const initStartDate = initialDate
+      ? new Date(initialDate)
+      : initialDate === ""
+      ? null
```
Comment (EdwinGuzman) [inline src/components/DatePicker/DatePicker.tsx:290]: I'm surprised that there's no ts issue with the `FullDateType` interface when setting `let initFullDate: FullDateType = { startDate: initStartDate };` since `initialStartDate` could be empty.

**Comment by EdwinGuzman** [comment]
This looks good. For the draft, can you add a working example in storybook with the initial empty value set? So we can see it working.

File: src/components/DatePicker/DatePicker.tsx (hunk 1 above)
Comment (oliviawongnyc) [reply src/components/DatePicker/DatePicker.tsx:290]: I'm not exactly sure why either, but I added the possibility of null to this interface.

File: src/components/DatePicker/DatePicker.stories.tsx [hunk 2]
```diff
@@ -187,13 +187,51 @@ export const OtherStates: Story = {
 };
 
 export const InitialDate: Story = {
+  render: () => (
```
Comment (EdwinGuzman) [inline src/components/DatePicker/DatePicker.stories.tsx]: This is great, thanks!

**Comment by bigfishdesign13** [comment]
I think the choices that you made for the prop functionality are fine.  But for fun, here is another possibility...

Omitting the `initialDate` or `initialDateTo` props or passing empty strings to the props would render empty fields.  To get today's date, pass a wildcard like `@today`.  This would be a new concept in our components, but it would then allow for other wildcards like `@yesterday`, `@nextweek`, etc.

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 that sounds like a good feature for another PR.

@oliviawongnyc I made a small update to the Storybook example. I'm using this now in a Research Catalog form and I wanted to make sure that an empty value does return null or an empty string. For example, If the user only wants to search from start year of 1950 until now, the end year can be empty and the `onChange` function will return the appropriate values (as far as I can tell). Based on this, this update looks good.

**Comment by oliviawongnyc** [comment]
> @oliviawongnyc I made a small update to the Storybook example. I'm using this now in a Research Catalog form and I wanted to make sure that an empty value does return null or an empty string. For example, If the user only wants to search from start year of 1950 until now, the end year can be empty and the `onChange` function will return the appropriate values (as far as I can tell). Based on this, this update looks good.

@EdwinGuzman, I'm not entirely sure what you mean. What do you expect the values to be if a user empties one of the inputs?

**Comment by EdwinGuzman** [comment]
@oliviawongnyc I expect it to be null or undefined.

**Comment by oliviawongnyc** [comment]
> @oliviawongnyc I expect it to be null or undefined.

@EdwinGuzman, gotcha. It's null. Do you feel like this is good to go?

**Review by EdwinGuzman — COMMENTED** [review]
This looks good to me but update the changelog for the repo and for the component, and add one test where the values passed are empty and no dates get rendered initially.

**Review by EdwinGuzman — APPROVED** [review]
